File: src/main/java/com/myudog/myulib/api/event/events/EntitySpawnEvent.java
package com.myudog.myulib.api.event.events;

import com.myudog.myulib.api.event.FailableEvent;
import net.minecraft.world.entity.Entity;
import net.minecraft.world.level.Level;

public class EntitySpawnEvent implements FailableEvent {
    private final Entity entity;
    private final Level Level;
    private String errorMessage;

    public EntitySpawnEvent(Entity entity, Level Level) {
        this(entity, Level, null);
    }

    public EntitySpawnEvent(Entity entity, Level Level, String errorMessage) {
        this.entity = entity;
        this.Level = Level;
        this.errorMessage = errorMessage;
    }

    public Entity getEntity() {
        return entity;
    }

    public Level getLevel() {
        return Level;
    }

    @Override
    public String getErrorMessage() {
        return errorMessage;
    }

    @Override
    public void setErrorMessage(String errorMessage) {
        this.errorMessage = errorMessage;
    }
}

File: src/main/kotlin/com/myudog/myulib/api/event/events/EntitySpawnEvent.kt
package com.myudog.myulib.api.event.events

import com.myudog.myulib.api.event.FailableEvent
import net.minecraft.entity.Entity
import net.minecraft.world.World

/**
 * 當任何實體準備生成到世界上時觸發。
 * 實作 FailableEvent，允許攔截者取消特定實體的生成 (CONSUME/FAILED)。
 */
data class EntitySpawnEvent(
    val entity: Entity,
    val world: World,
    override var errorMessage: String? = null
) : FailableEvent